packages/block-library/src/buttons/edit.native.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import {
+	InnerBlocks,
+	BlockControls,
+	AlignmentToolbar,
+} from '@wordpress/block-editor';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import { createBlock } from '@wordpress/blocks';
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { name as buttonBlockName } from '../button/';
+import styles from './editor.scss';
+
+const ALLOWED_BLOCKS = [ buttonBlockName ];
+const BUTTONS_TEMPLATE = [ [ 'core/button' ] ];
+const ALIGNMENT_MAP = {
+	left: 'flex-start',
+	center: 'center',
+	right: 'flex-end',
+};
+
+function ButtonsEdit( {
+	isSelected,
+	attributes,
+	setAttributes,
+	onDelete,
+	onAddNextButton,
+	shouldDelete,
+	isParentSelected,
+	isSelectedButton,
+} ) {
+	const { align } = attributes;
+	const [ maxWidth, setMaxWidth ] = useState( 0 );
+	const shouldRenderFooterAppender = isSelected || isSelectedButton;
+
+	function updateAlignment( nextAlign ) {
+		setAttributes( { align: nextAlign } );
+	}
+
+	function renderAppender() {
+		if ( shouldRenderFooterAppender ) {
+			return (
+				<InnerBlocks.ButtonBlockAppender
+					flex={ false }
+					customOnAdd={ onAddNextButton }
+				/>
+			);
+		}
+		return null;
+	}
+
+	function onLayout( { nativeEvent } ) {
+		const { width } = nativeEvent.layout;
+		const { marginLeft: nestedSpacing } = styles.nestedButtons;
+		const parentWidth =
+			width + ( isParentSelected ? 2 * nestedSpacing : 0 );
+
+		setMaxWidth( parentWidth );
+	}
+
+	const buttonsStyle = {
+		justifyContent: ALIGNMENT_MAP[ align ],
+	};
+
+	return (
+		<>
+			<BlockControls>
+				<AlignmentToolbar
+					isCollapsed={ false }
+					value={ align }
+					onChange={ updateAlignment }
+				/>
+			</BlockControls>
+			<View onLayout={ onLayout }>
```

Comment:
How about introduce your resizeObserver() hook here ? Or we should refactor later ?